Sources/StreamChatUI/ChatMessageList/ChatMessageListView+DiffKit.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import DifferenceKit
+import StreamChat
+import UIKit
+
+extension ChatMessageListView {
+    internal func reloadMessages(
+        previousSnapshot: [ChatMessage],
+        newSnapshot: [ChatMessage],
+        with animation: @autoclosure () -> RowAnimation,
+        completion: (() -> Void)? = nil
+    ) {
+        let changeset = StagedChangeset(
+            source: previousSnapshot,
+            target: newSnapshot
+        )
+        // This is need because DiffKit doesn't provide a completion block for when the reload is finished.
+        // The CATransaction notifies when animations are finished executing between begin() and commit().
+        CATransaction.begin()
+        CATransaction.setCompletionBlock(completion)
+        reload(
+            using: changeset,
+            with: animation()
+        ) { [weak self] newMessages in
+            self?.onNewDataSource?(newMessages)
+        }
+        CATransaction.commit()
+    }
+}
+
+extension ChatMessage: Differentiable {
+    public func isContentEqual(to source: ChatMessage) -> Bool {
```

Comment:
Just a note: This can actually be quite expensive right now

Reply:
Yes, this is not perfect right now, because it runs some lazy properties which hit the DB. Still, for most use cases we have really nice performance. But when the message list is being completely stressed (1000+ messages and the stress test running) it drops a little bit the FPS, but currently, we don't have another option to improve this.